lib/mp4/transmuxer.js

```diff
@@ -891,17 +892,10 @@ CoalesceStream.prototype.flush = function(flushSource) {
 
   this.emittedTracks += this.pendingTracks.length;
 
-  initSegment = mp4.initSegment(this.pendingTracks);
+  event.serializedTracks = mp4Utils.serializeTracks(this.pendingTracks);
```

Comment:
You're definitely right that any manipulation that is done for dealing with the web worker should be in 'transmux-worker' in videojs-contrib-media-sources. I moved it there.

Right now I'm still doing the objToTypedArr and typedArrToObj in videojs-contrib-media-sources, as I still wanted the other information from the tracks object (except samples, which I delete to save space on the copy), and it seemed simpler than doing the zero copy for multiple conditionally existing inner sps and pps values (re: 2), and retains the offset values (re: 1). Though I could be wrong on which is simpler/more straightforward. Let me know what you think on that PR!